CharlieBackend.AdminPanel/Startup.cs

```diff
@@ -33,7 +34,28 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddHttpContextAccessor();
 
-            services.AddTransient<IHttpUtil, HttpUtil>();
+            services.AddScoped<IHttpUtil, HttpUtil>((ctx) =>
+            {
+                IHttpContextAccessor httpContext = ctx.GetService<IHttpContextAccessor>();
+
+                string accessToken = httpContext.HttpContext.Request.Cookies["accessToken"];
+
+                if (accessToken != null)
+                {
+                    IDataProtectionProvider provider = ctx.GetService<IDataProtectionProvider>();
+
+                    IDataProtector protector = provider.CreateProtector(config.Cookies.SecureKey);
+
+                    string unprotectedToken = protector.Unprotect(accessToken);
+
+                    return new HttpUtil(config.Urls.Api.Https, unprotectedToken);
```

Comment:
You can have one constructor that takes IHttpContextAccessor and IConfiguration and then extract all data in the construction.
Then depending if there is a token add it to _client.

Reply:
Startup is [restored](https://github.com/ita-social-projects/WhatBackend/pull/362/commits/7cee80a627ae3fca48e7fe2124ab075e3320dde3)
Added HttpUtil [constructor](https://github.com/ita-social-projects/WhatBackend/pull/362/commits/bde6dbbe8334e2be6a669da9eaab55e50e00e42d).